### DISCUSSION TITLE
Calculating the working day in sql sqlitestudio

### DISCUSSION THREAD

**Discussion by PatriaMinas** [pawelsalawa/sqlitestudio#4406] [Show and tell]
Hi,
How to know the working day between a date and time using sql sqlitestudio?
I hope this can help!

Portuguese translation:
"Olá,
Como saber o dia útil entre uma data e o tempo utilizando sql sqlitestudio?
Eu espero que isso possa ajudar!"

![Captura de tela de 2022-03-01 19-23-34](https://user-images.githubusercontent.com/95147314/156259733-25b87c84-cea9-4c81-8725-da432c5eb4a7.png)

![Captura de tela de 2022-03-01 19-36-27](https://user-images.githubusercontent.com/95147314/156260744-d895f361-cfe9-4cef-bbc2-35d0252c5cb9.png)

The script in text:

CREATE TABLE CALEND(
CLDAY DATE PRIMARY KEY UNIQUE, 
NWEEK INT (1) AS (strftime('%w', CLDAY)), 
NHOLI INT (1) AS (CASE WHEN strftime('%m-%d', CLDAY) IN ('01-01', '12-25'/*more from your country*/)
                     THEN 1 ELSE 0 END), 
DWORK INT (1) AS ((CASE WHEN NWEEK IN (0, 6) OR NHOLI = 1 THEN 0 ELSE 1 END)))
;
--Import csv file (calendar.csv) 1900-01-01 to 2099-12-31 example, unique column.
;
--Create sqlitestudio function dweeknd() 
SELECT MAX(Q.CLDAY) CLDAY
FROM (
SELECT L.CLDAY
FROM CALEND L 
WHERE L.CLDAY >= @DATE1 AND L.DWORK = 1 LIMIT (@TIME2 + 1)) Q
;

SELECT dweeknd('2022-01-01', 365) DAYWORKN
;

[calendar.csv](https://github.com/pawelsalawa/sqlitestudio/files/8165188/calendar.csv)

**Comment by pawelsalawa** [comment]
So... do I understand you correctly that you are sharing with everybody a script for Custom SQL Function?

**Reply by PatriaMinas** [reply]
Correct. The problems that I can solve through SqliteStudio I am publishing here.
So, I hope to help other people who have the same problem to develop a script.
Please, if this isn't the place for this post, I'll do it in the right space.

**Reply by pawelsalawa** [reply]
This is fine. I was just making sure you're posting solution and not some question.
